src/mapwindow.jl

```diff
@@ -273,13 +365,100 @@ end
 # This is slightly faster than a circular buffer
 @inline cyclecache(b, x) = b[1], (Base.tail(b)..., x)
 
+
 replace_function(f) = f
-replace_function(::typeof(median!)) = function(v)
-    inds = indices(v,1)
-    Base.middle(Base.select!(v, (first(inds)+last(inds))÷2, Base.Order.ForwardOrdering()))
+
+replace_function(::typeof(median!)) = function(v,m_histogram,mode,window)
+   # Handle the boundary points with mode -1
+    if(mode==-1)
+        inds = indices(v,1)
+        return Base.middle(Base.select!(v, (first(inds)+last(inds))÷2, Base.Order.ForwardOrdering()))
+    else
+        
+        window_size=size(v,1)
+        dims = map(x->x.stop-x.start+1,window)
+        width=window[1].stop-window[1].start+1
+        inds = indices(v,1)
+        if mode == 0
+        # Update the histogram according to new entries
+            for i = first(inds):last(inds)
+                id= trunc(Int64,(v[i]*255))+1
+                m_histogram[id]+=1
+            end
+            counter=0
+
+            for i =1:256
+                counter+=m_histogram[i]
+            end
+        # Compute the median
+            tempsum = 0
+            m_index=-1
+            for i = 1:256
```

Comment:
A lot of this hard-codes the fact that `histogram` has length 256, but note that `histogram` is not even allocated in this function so people changing code in one place might cause bugs in a different place.

`for in in linearindices(histogram)` would be more general.